diff --git a/src/libcamera/pipeline/raspberrypi/raspberrypi.cpp b/src/libcamera/pipeline/raspberrypi/raspberrypi.cpp
index 00600441..994cd05e 100644
--- a/src/libcamera/pipeline/raspberrypi/raspberrypi.cpp
+++ b/src/libcamera/pipeline/raspberrypi/raspberrypi.cpp
@@ -1730,8 +1730,8 @@ int RPiCameraData::configureIPA(const CameraConfiguration *config, ipa::RPi::IPA
 int RPiCameraData::loadPipelineConfiguration()
 {
 	config_ = {
-		.minUnicamBuffers = 2,
-		.minTotalUnicamBuffers = 4,
+		.minUnicamBuffers = 0,
+		.minTotalUnicamBuffers = 1,
 		.disableStartupFrameDrops = false,
 		.unicamTimeoutValue = 0,
 	};
